connection.go

```diff
@@ -232,6 +236,14 @@ func (me *Connection) LocalAddr() net.Addr {
 	return &net.TCPAddr{}
 }
 
+// ConnectionState returns basic TLS details of the underlying transport.
+func (me *Connection) ConnectionState() tls.ConnectionState {
```

Comment:
This function can confuse the reader slightly: it returns a `tls.ConnectionState` but AMQP 0-9-1 connections also have state and so `*Connection.ConnectionState` doesn't really indicate that what's returned in a TLS upgrade state.

Please rename it to `TLSConnectionState`.